### ISSUE TITLE
Incorrect VHDL component documentation

### ISSUE THREAD

**Issue by jtonfat** [doxygen/doxygen#7604]
Hi!,
I found the following issue:
When more than one component is declared in an architecture, the second component uses the last port doxygen comment. (See line 40)

The same issue happens with the doxygen documentation of instantiation. In this example (see line 56), the last port doxygen comment of Component MainReset is assigned to the instatiation of PowerReset. 

I think that a solution could be to ignore doxygen comments inside Component declaration . 

I attach the example and doxygen config file. The issue appears in version 1.8.17.

[bug_vhdl_components.zip](https://github.com/doxygen/doxygen/files/4235640/bug_vhdl_components.zip)

**Comment by albert-github** [comment]
Looks to me that ports within the documentation of components are not supported. When not documenting these ports with `--!` but with just `--` it looks to me that this problem is solved, probably better it would be to have ports supported with components as well.

**Comment by albert-github** [comment]
Had a quick look in the `vhdlparser.jj` and it looks to me that the solution lies in part:
```
void component_declaration() :  {QCString s;}
{
 <COMPONENT_T> s=identifier() [ <IS_T> ]
   { m_sharedState->currP=VhdlDocGen::COMPONENT; }
  [ generic_clause() ]
  [ port_clause() ]
   {
     outlineParser()->addVhdlType(s.data(),outlineParser()->getLine(COMPONENT_T),Entry::VARIABLE_SEC,VhdlDocGen::COMPONENT,0,0,Public);
     m_sharedState->currP=0;
  }
  <END_T> <COMPONENT_T> [ identifier() ] <SEMI_T>

}
```
tat probably should read more loke:
```
void component_declaration() :  {QCString s;}
{
 <COMPONENT_T> s=identifier() [ <IS_T> ]
 [  { m_sharedState->currP=VhdlDocGen::GENERIC;m_sharedState->parse_sec=GEN_SEC; } generic_clause()]
 [ { m_sharedState->currP=VhdlDocGen::PORT; } port_clause()]
  <END_T> <COMPONENT_T> [ identifier() ] <SEMI_T>

}
```
but the problem that remains here is that the component ports are still on a level to high i.e. there should probably be a level `Components` in `BasicControl_str`.

Note: I'm also a bit doubtful (in the documentation) that `ieee` and `std_logic_1164` are documented under the entity `BasicControl` and that the architecture `BasicControl_str` is mentioned unde Entities.

@maddox11 do you have an idea?

**Comment by jtonfat** [comment]
> Looks to me that ports within the documentation of components are not supported. When not documenting these ports with `--!` but with just `--` it looks to me that this problem is solved, probably better it would be to have ports supported with components as well.

I am not sure if supporting component ports will be a good solution since in anycase there should be an entity that match the component declaration and in this entity the ports should be documented.

**Comment by albert-github** [comment]
(as a disclaimer I'm not a vhdl programmer)

@jtonfat In your example I don't see all ports in entities , or is this just because of example is a bit smaller?
So you suggest that my solution (i.e. `--!` should be `--`) for components ports is OK or would you have the documentation in the "component ports" merged with the "entity ports"?

**Comment by jtonfat** [comment]
Yes, I think your solution is ok. However, I don't know if it's possible to ignore the doxygen comments for the ports in the component.

**Comment by maddox11** [comment]
a  possible workaround 
  -- Component MainReset
   --! \brief  generate the main reset for the global logic and the SPWcodec cores.
   --! \param  port BOARD_REG_RST  SPW1 register-triggered reset.
   --! \param  port SPW1_REG_RST   SPW2 register-triggered reset.
   --!    .........
